clustering/common/src/main/java/org/jboss/as/clustering/controller/CommonUnaryRequirement.java

```diff
@@ -5,36 +5,33 @@
 
 package org.jboss.as.clustering.controller;
 
-import java.security.KeyStore;
-
-import javax.net.ssl.SSLContext;
-import javax.sql.DataSource;
-
+import org.jboss.as.controller.services.path.PathManager;
 import org.jboss.as.network.OutboundSocketBinding;
 import org.jboss.as.network.SocketBinding;
 import org.wildfly.clustering.service.UnaryRequirement;
-import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.service.descriptor.ServiceDescriptor;
 
 /**
  * Enumerates common unary requirements for clustering resources
  * @author Paul Ferraro
  */
+@Deprecated(forRemoval = true)
```

Comment:
It is strongly recommended that the reason for deprecating a program element be explained in the documentation, using the @deprecated javadoc tag. The documentation should also suggest and link to a recommended replacement API, if applicable. A replacement API often has subtly different semantics, so such issues should be discussed as well.